tests/test_cos_integration.py

```diff
@@ -52,110 +52,59 @@ async def test_build_and_deploy_istio_charms(ops_test: OpsTest):
         raise_on_blocked=False,
         timeout=90 * 10,
     )
-
-
-async def test_prometheus_grafana_integration_istio_pilot(ops_test: OpsTest):
-    """Deploy prometheus and required relations, then test the metrics."""
-    await ops_test.model.deploy(
-        PROMETHEUS_K8S,
-        channel=PROMETHEUS_K8S_CHANNEL,
-        trust=PROMETHEUS_K8S_TRUST,
+    # Deploying grafana-agent-k8s and add all relations
+    await deploy_and_assert_grafana_agent(
+        ops_test.model, ISTIO_PILOT, metrics=True, dashboard=True, logging=False
```

Comment:
dashboard=false?

Reply:
For pilot we need to have it, but you are right for istio-gateway it should be removed.